### ISSUE TITLE
Vendored code inside of vendored code

### ISSUE THREAD

**Issue by bkrahmer** [gogradle/gogradle#161]
Do you support the vendored inside of vendored use case?  I'm starting my project from scratch, and upon hitting an entry like this in our vendor/vendor.json file:
`
                {
                        "checksumSHA1": "Uzyon2091lmwacNsl1hCytjhHtg=",
                        "origin": "github.com/hashicorp/consul/vendor/github.com/hashicorp/go-cleanhttp",
                        "path": "github.com/hashicorp/go-cleanhttp",
                        "revision": "a6bb339621daadf20eea29e5a68afd7e39e3e239",
                        "revisionTime": "2016-12-01T05:47:33Z"
                }
`

I get this error:
`
Caused by: java.lang.IllegalStateException: Sorry, we should not be here: {name=github.com/hashicorp/consul, version=a6bb339621daadf20eea29e5a68afd7e39e3e239}
        at com.github.blindpirate.gogradle.task.GoInit.mapEntryToString(GoInit.java:152)
        at com.github.blindpirate.gogradle.task.GoInit.appendOneLine(GoInit.java:135)
        at com.github.blindpirate.gogradle.task.GoInit.lambda$appendToBuildDotGradle$2(GoInit.java:124)
        at com.github.blindpirate.gogradle.task.GoInit.appendToBuildDotGradle(GoInit.java:124)
        at com.github.blindpirate.gogradle.task.GoInit.initByExternalLockfiles(GoInit.java:114)
        at com.github.blindpirate.gogradle.task.GoInit.init(GoInit.java:81)
`

**Comment by blindpirate** [comment]
Thanks for reporting, that's definitely a missing test case of Gogradle. I'll fix it in December release. As a workaround, you could put this line in your `build.gradle`:

```
build name:'github.com/hashicorp/go-cleanhttp', host: [name:'github.com/hashicorp/consul', version:'a6bb339621daadf20eea29e5a68afd7e39e3e239'], vendorPath: 'vendor/github.com/hashicorp/go-cleanhttp', transitive: false
```

**Comment by blindpirate** [comment]
Fixed by `0.8.0`. Any feedbacks are welcomed.
